### PR TITLE
fix: Read data value set using dataViewOrgUnits [DHIS2-10864]

### PR DESCRIPTION
**Pull request by abyot**

### CODE REVIEW

**Review by jbee — COMMENTED** [review]
I find it odd to have both a cached and an uncached version available in the service API. Caching should be transparent or it appears to be wrong solutions if it does matter. Also I wonder if this shouldn't be a DB operation instead. If this is so "heavy" or dependent on relations it maybe should be answered by a dedicated DB query instead which also should remove the need for caching I guess.

**Review by ameenhere — APPROVED** [review]
Agree with @jbee , had a similar discussion with @enricocolasante a while ago. We eventually decided to "do it the same way" for the time being until we come to an agreement on how we want to re-implement this.

**Comment by abyot** [comment]
@ameenhere, @jbee, I was tempted to go with the caching only but then thought doing it both ways gives us flexibility and followed the trend

**Comment by larshelge** [comment]
I am not sure if there is a perfect answer to the caching issue above. In some cases it is not feasible in terms of performance to maintain a cache which is never stale. It could make sense to offer methods where the caller can do the consideration on the performance/freshness trade-off, and e.g. say that based on the relevant use-case, it is acceptable to get a response which may be slightly stale in order to get the performance benefit.

**Review by jbee — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/organisationunit/DefaultOrganisationUnitService.java [hunk 1]
```diff
@@ -496,6 +499,42 @@ public boolean isInUserHierarchy( User user, OrganisationUnit organisationUnit )
         return organisationUnit.isDescendant( user.getOrganisationUnits() );
     }
 
+    @Override
+    @Transactional( readOnly = true )
+    public boolean isInUserViewHierarchy( OrganisationUnit organisationUnit )
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/organisationunit/DefaultOrganisationUnitService.java:504]: Can we rename to `isInUserDataViewHierarchy` here to be consistent with `User#dataViewOrganisationUnits` @abyot ?

Comment (abyot) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/organisationunit/DefaultOrganisationUnitService.java:504]: @larshelge, sure we can do.

**Pull request merged**